src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -492,6 +492,26 @@ parseNumberBinder = NumberBinder <$> (sign <*> number)
          <|> (symbol' "+" >> return id)
          <|> return id
 
+parseTypedBinder :: TokenParser Binder
+parseTypedBinder = do
+  b <- typeableBinder
+  indented *> doubleColon
+  t <- parseType
+  return $ TypedBinder t b
+    where typeableBinder = P.choice (map P.try
```

Comment:
It might be better here to parse any binder, then assert it matches a predicate `isTypeableBinder`, so that you can give a better error message.